### PR TITLE
Bump code to monero-rpc-rs and bump Monero clients to 0.18.0.0

### PR DESCRIPTION
**Pull request by LeoNero**
This bumps Monero clients to 0.18.0.0 like #614 does. If the other PR gets merged first, I'll update this one.

### CODE REVIEW

File: tests/swap.rs [hunk 1]
```diff
@@ -832,7 +832,7 @@ async fn run_restore_checkpoint_bob_pre_buy_alice_pre_buy(
     drop(monero_wallet_lock);
     drop(lock);
     let delta_balance = after_balance.balance - before_balance.balance;
-    assert!(delta_balance > 999660000000);
+    assert!(delta_balance.as_pico() > 999660000000);
```
Comment (TheCharlatan) [inline tests/swap.rs:835]: NIt: I'd prefer doing the conversion on the other side, so we don't change the data or type to be tested.

**Comment by TheCharlatan** [comment]
I'm surprised this did not lead to more type improvements in the monero syncer, but we can figure out how to handle the remaining raw types at a later stage.

Anyway, nice work and I am glad this went in so seemingly painless compared to our Bitcoin version bump that we are currently working on. ACK once the nit is addressed or further rationalized.

**Comment by LeoNero** [comment]
@TheCharlatan addressed.

**Review by TheCharlatan — APPROVED** [review]

**Comment by h4sh3d** [comment]
LGTM, will rebase #614

**Comment by h4sh3d** [comment]
Oh, I see now that you didn't updated the CI containers in `.github/workflows/test.yml` @LeoNero 

cc: @TheCharlatan

**Comment by TheCharlatan** [comment]
Oh damn, I missed that as well :O

**Comment by LeoNero** [comment]
oh will update!

**Comment by LeoNero** [comment]
I think my vim extension that searches words in the files is missing `.` folders

**Pull request merged**